### ISSUE TITLE
Improve error from dynamic ports

### ISSUE THREAD

**Issue by espenfl** [aiidateam/aiida_core#1980]
When working with workchains sometimes one does not get things right (well, in the context that one does not want to use dynamic mode) and get this:

`ValueError: Invalid type <class 'aiida.common.extendeddicts.AttributeDict'> for dynamic port value: expected (<class 'aiida.orm.data.Data'>, <class 'aiida.orm.implementation.django.calculation.Calculation'>)`

It would be great to also specify that this is an `input` or `output`, in addition to what the name of the node was. Would this be possible? Would make it a bit easier to debug etc.

**Comment by sphuber** [comment]
This is actually code that resides in the `plumpy` dependency, so this will have to be resolved there. The relevant code is in `plumpy.ports` and part looks like:
```
    def validate_dynamic_ports(self, port_values=None):
        """
        Validate port values with respect to the dynamic properties of the port namespace. It will
        check if the namespace is actually dynamic and if all values adhere to the valid types of
        the namespace if those are specified

        :param port_values: an arbitrarily nested dictionary of parsed port values
        :return: valid or not, error string|None
        :rtype: tuple(bool, str or None)
        """
        is_valid, message = True, None

        if port_values and not self.dynamic:
            return False, 'Unexpected ports {}, for a non dynamic namespace'.format(
                port_values)

        if self._valid_type is not None:
            valid_type = self._valid_type
            for port_name, port_value in port_values.items():
                if not isinstance(port_value, valid_type):
                    return False, 'Invalid type {} for dynamic port value: expected {}'.format(
                        type(port_value), valid_type)

        return is_valid, message
```

We have access to the `port_name`, which is the key under which the user passed the value and that could be included in the exception message.

**Comment by sphuber** [comment]
This is now fixed in `plumpy==0.12.1` and the dependency has been updated in PR #2324 
Error message is now for example:
```
ValueError: Error occurred validating port 'inputs': Unexpected ports {'b': 2}, for a non dynamic namespace
```
